org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/ProjectNatureUpdater.java

```diff
@@ -48,7 +51,15 @@ public static void update(IProject project, Optional<List<OmniEclipseProjectNatu
 
         Result<String> result = ManagedModelMergingStrategy.calculate(existingNatures, modelNatures, managedNatures);
 
-        description.setNatureIds(result.getNextElements().toArray(new String[0]));
+        String[] natureIds = FluentIterable.from(result.getNextElements()).filter(new Predicate<String>() {
```

Comment:
⭕️ How is this related to the refactoring?

Reply:
This was a temporary fix for the broken `ImportingWtpProjectsTest`. The problem was already there as we set the unrecognized  WTP nature on the test project, but when we used the `Job` implementation the problem was swallowed. You can still find it in the logs: 

https://builds.gradle.org/viewLog.html?buildId=9843560&buildTypeId=Tooling_Master_IntegrationTests_Linux_Eclipse47IntegrationTests&tab=buildLog&_focus=2465#_state=61,1374

I've restored the original code and moved the fix to the `WorkspaceOperationsDelegate` test fixture.